framework/src/org/apache/cordova/CordovaWebViewImpl.java

```diff
@@ -500,6 +500,7 @@ public void onPageFinishedLoading(String url) {
 
             clearLoadTimeoutTimer();
 
+            cordova.onCordovaInit(pluginManager);
```

Comment:
I'm not sure it makes sense to wait until onPageFinished to dispatch onCordovaInit. Why not dispatch at the end of the .init() function?

If the concern is that you want to be able to communicate with the JS, I don't think onPageFinished guarantees that. Instead, you should have your plugin wait for an `exec()` to come in from the JS in order to communicate with it on start-up.